workflow/hydrator/hydrator.go

```diff
@@ -101,12 +102,17 @@ func (h hydrator) Dehydrate(wf *wfv1.Workflow) error {
 	}
 	if packer.IsTooLargeError(err) || alwaysOffloadNodeStatus {
 		var offloadVersion string
-		err := waitutil.Backoff(writeRetry, func() (bool, error) {
```

Comment:
it is impossible for this to happen AFAIK - alwaysOffloadNodeStatus must always false , so the error must be IsTooLargeError